components/activedirectory/active_directory.go

```diff
@@ -0,0 +1,130 @@
+package activedirectory
+
+import (
+	"fmt"
+	"github.com/DataDog/test-infra-definitions/common"
+	"github.com/DataDog/test-infra-definitions/common/config"
+	"github.com/DataDog/test-infra-definitions/common/namer"
+	"github.com/DataDog/test-infra-definitions/components"
+	"github.com/DataDog/test-infra-definitions/components/command"
+	"github.com/DataDog/test-infra-definitions/components/os"
+	"github.com/DataDog/test-infra-definitions/components/remote"
+	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/pulumiverse/pulumi-time/sdk/go/time"
+)
+
+// Output is an object that models the output of the resource creation
+// from the Component.
+// See https://www.pulumi.com/docs/concepts/resources/components/#registering-component-outputs
+type Output struct {
+	components.JSONImporter
+}
+
+// Component is an Active Directory domain component.
+// See https://www.pulumi.com/docs/concepts/resources/components/
+type Component struct {
+	pulumi.ResourceState
+	components.Component
+	namer namer.Namer
+	host  *remote.Host
+}
+
+// Export registers a key and value pair with the current context's stack.
+func (dc *Component) Export(ctx *pulumi.Context, out *Output) error {
+	return components.Export(ctx, dc, out)
+}
+
+// A little structure to help manage state for the Active Directory component
+type activeDirectoryContext struct {
+	pulumiContext    *pulumi.Context
+	comp             *Component
+	timeProvider     *time.Provider
+	createdResources []pulumi.Resource
+}
+
+// NewActiveDirectory creates a new instance of an Active Directory domain deployment
+// Example usage:
+//
+//	activeDirectoryComp, activeDirectoryResources, err := activedirectory.NewActiveDirectory(pulumiContext, awsEnv.CommonEnvironment, host,
+//		activedirectory.WithDomainController("datadogqa.lab", "Test1234#"),
+//	    activedirectory.WithDomainUser("datadogqa.lab\\ddagentuser", "Test5678#"),
+//	)
+//	if err != nil {
+//		return err
+//	}
+//	err = activeDirectoryComp.Export(pulumiContext, &env.ActiveDirectory.Output)
+//	if err != nil {
+//		return err
+//	}
+func NewActiveDirectory(ctx *pulumi.Context, e *config.CommonEnvironment, host *remote.Host, options ...Option) (*Component, []pulumi.Resource, error) {
+	if host.OS.Descriptor().Family() != os.WindowsFamily {
+		// Print flavor in case OS family don't match, as it's more precise than the family (and the .String() conversion already exists).
+		return nil, nil, fmt.Errorf("wrong Operating System family, expected Windows, got %s", host.OS.Descriptor().Flavor.String())
+	}
+
+	params, err := common.ApplyOption(&Configuration{}, options)
+	if err != nil {
+		return nil, nil, err
+	}
+	// Make the resource options that will be passed to the component
```

Comment:
<!-- Suggestions propose improvements to the current subject. It’s important to be explicit and clear on what is being suggested and why it is an improvement. -->
**💬 suggestion**
```suggestion
	// Tell the component its parent is host and allow to delete it with host. This will prevent uninstalling AD on host delete, we will directly delete the host
```